LukeMason33/CharityLocal
# Feature/dashboard

### Is this a fix or a feature?
- This is a feature that contains the finishing builds for the components that make up the dashboard after a user searches for a charity. The components that were built in this branch are : 
    - `MainDashboard.js` - this component is what gets rendered when the 'Find Charities' button is clicked on the landing page. This component contains a header for user navigation in between pages, and also renders our `CharityContainer` component.
    - `CharityContainer.js` - this component gets rendered from the `MainDashboard` component. It takes in the prop of `charities` that's passed from `App` component, to the `MainDashboard` component, to here. This component then renders the `CharityCard` component for each charity that matches the user search. It passes down a `charity` prop to the `CharityCard` component for each charity. 
    - `CharityCard.js` - this component gets rendered from the `CharityContainer` component. This component takes in a `charity` prop, which is used to display unique information for a specific charity. The charity name acts as a link to the charity's official website.
- I also added in a loading screen for when the fetch is retrieving our data, as well as an error message that will be displayed on the landing page if there are no charities returned from the fetch request. This is all done in `App.js`.

### What is the change/fix?
- The change is listed above.

### Where should the reviewer start?
- The reviewer should start in `App.js` and look at how this component is being rendered by a `Route` component, examine the `loading` as well as the `error` hook (which is passed to our `LandingPageForm` component to display the error message there).  Then the reviewer should go to `MainDashboard.js` and examine how this component is rendering our `CharityContainer` component. Next, the reviewer should then jump into `CharityContainer.js` and examine the method in here which returns a `CharityCard` component for each charity that the user search matches. From here, the reviewer should then go to `CharityCard.js` and examine how each charities information is being displayed. 

### How should this be tested?
- This should be tested by opening up the app and inputting a category, state, and valid city for that state. After clicking the 'Find Charities' button, a loading screen should display and then the main dashboard should display the charities that matched the search. After successfully doing this, go back to the landing page and then do fill out the category input, state input, and type random letters in the city input. After clicking the 'Find Charities' button, should be redirected to the landing page with an error message.
